src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -43,7 +50,10 @@ ReactRef.attachRefs = function(instance, element) {
   }
 };
 
-ReactRef.shouldUpdateRefs = function(prevElement, nextElement) {
+ReactRef.shouldUpdateRefs = function(
+  prevElement: ReactElement | null | false,
+  nextElement: ReactElement | null | false,
```

Comment:
Our unit tests do have examples where it can be `number` and `string` indeed.

So sorry for the churn here, I'm going to try to be more careful next time.